app/api_clients/azure.py

```diff
@@ -0,0 +1,85 @@
+import logging
+
+from azure.core.exceptions import (
+    AzureError,
+    ClientAuthenticationError,
+    ResourceExistsError,
+    ResourceNotFoundError,
+)
+from azure.identity import DefaultAzureCredential
+from azure.storage.blob import BlobServiceClient
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+
+AZURE_SA_CREDENTIALS = DefaultAzureCredential()
+
+
+class AzureBlobServiceClient:
+    def __init__(
+        self,
+        container_name: str,
+        connection_string: str | None = None,
+    ):
+        if connection_string:
+            self.blob_service_client = BlobServiceClient(
```

Comment:
you should use the async version of such client (https://learn.microsoft.com/es-es/python/api/azure-storage-blob/azure.storage.blob.aio.blobserviceclient?view=azure-python)